### DISCUSSION TITLE
Postage rate API

### DISCUSSION THREAD

**Discussion by gshaw** [chitchats/chitchats-api-doc#21] [General]
There have been [various questions](https://github.com/chitchats/chitchats-api-doc/issues/3#issuecomment-747112633) around the postage rate API.  I'm going to lock that issue and move the discussion here:

To get a rate using the Chit Chats API you need to create a shipment with an unknown postage type.  The response will return a shipment object with an array of rates.

The ideal workflow would be to create the shipment for the order that is in the customer's cart and then show the rates from that shipment. If check out is completed, buy the postage for that shipment, if the cart is abandoned you can delete or ignore the shipment. This way the rate that was shown is guaranteed to be the rate you've shown.

I can appreciate that it seems odd that you need to create a shipment query rates.

One of the reasons for this is that if you get a rate and display it to the customer but then later create a new shipment and buy the postage there isn't any guarantee that the rates will be the same. Rates don't change often but they do change.

**Comment by wesbos** [comment]
@gshaw thanks for your replies here! Makes sense to me!

**Reply by Medanaresh** [reply]
Hi Dev Team, Good Morning.

for getting postage we hit the create shipment API with postage type as unknown but we are getting empty rates object, and shipment is created.

kindly find the below screenshots
![image](https://user-images.githubusercontent.com/14160818/112156966-c0f96c80-8c0c-11eb-94d8-eac335156cc9.png)
![image](https://user-images.githubusercontent.com/14160818/112157065-d79fc380-8c0c-11eb-98aa-f72a1390dc4b.png)

**Reply by gshaw** [reply]
This reply is a dupe of this issue: https://github.com/chitchats/chitchats-api-doc/issues/43
